### PR TITLE
POSIX Debugging

### PR DESCRIPTION
**Pull request by zrax**
Some additional fixes for debugging stuff on GCC/POSIX

### CODE REVIEW

File: cmake/CompilerChecks.cmake [hunk 1]
```diff
@@ -11,6 +11,13 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
     endif()
 endif()
 
+# MSVC automatically defines -D_DEBUG when /MTd or /MDd is set, so we
+# need to make sure it gets added for other compilers too
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_DEBUG")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_DEBUG")
+endif()
```
Comment (dpogue) [inline cmake/CompilerChecks.cmake]: Under what cases would this not be set? Wouldn't this be controlled by the CMake Build Type (Debug, Release, RelWithDebInfo, etc.)?

Comment (zrax) [inline cmake/CompilerChecks.cmake]: Oops, I meant to add that to the _DEBUG version of the flags...

Comment (zrax) [inline cmake/CompilerChecks.cmake]: Fixed

File: Sources/Plasma/PubUtilLib/plFile/plSecureStream.cpp [hunk 2]
```diff
@@ -325,9 +325,10 @@ uint32_t plSecureStream::IRead(uint32_t bytes, void* buffer)
         }
         else
         {
-            // TODO: errno equivalent for *nix
 #if HS_BUILD_FOR_WIN32
             hsDebugMessage("Error on Windows read", GetLastError());
+#else
+            hsDebugMessage("Error on POSIX read", errno);
```
Comment (dpogue) [inline Sources/Plasma/PubUtilLib/plFile/plSecureStream.cpp]: Does this require `#include <errno.h>`?

Comment (zrax) [inline Sources/Plasma/PubUtilLib/plFile/plSecureStream.cpp]: It didn't when I compiled it, but it probably doesn't hurt to include it for compliance.  Fixed

**Comment by dpogue** [comment]
:+1:

**Pull request merged**